aten/src/ATen/native/cuda/DilatedMaxPool2d.cu

```diff
@@ -279,12 +299,18 @@ void max_pool2d_with_indices_backward_out_cuda_template(
     outputHeight, outputWidth,
     /*cuda=*/ true);
 
-  const Tensor gradOutput = gradOutput_.contiguous();
+  const Tensor gradOutput = gradOutput_.contiguous(memory_format);
+
+  const int64_t out_stride_c = gradOutput.stride(-3);
+  const int64_t out_stride_h = gradOutput.stride(-2);
+  const int64_t out_stride_w = gradOutput.stride(-1);
+
   gradInput.resize_as_(input);
+  gradInput.unsafeGetTensorImpl()->empty_tensor_restride(memory_format);
 
   int64_t count = input.numel();
   dim3 grid;
-  int imgcount = inputWidth * inputHeight;
+  int imgcount = inputHeight * inputWidth;
   const int blocks = (imgcount + BACKWARD_THREADS - 1) / BACKWARD_THREADS;
```

Comment:
You can notice how they are binding block_x and grid_x towards spatial dimension here (because the old kernel handling NCHW, where spatial dimension is the fast changing dimension). This allows better memory accessing pattern. Backward for pooling is quite tricky to get the perf out, we can mimic their launch config here (but optimize it for NHWC stride)